### PR TITLE
Support new multi-chain wallet functionality

### PR DESCRIPTION
**Pull request by justinnout**
### 📦 Pull Request

The current implementation of magic-admin-js for metadata retrieval returns the Ethereum public_address.

We are adding functionality to query wallet(s) for any chain tied to the end-user.

Example (queries all Solana wallets created for the user if applicable):
`const result = await sdk.users.getMetadataByIssuerAndWallet('did:ethr:0x1234', WalletType.SOLANA);`

Example (queries all wallets created for the user if applicable):
`const result = await sdk.users.getMetadataByIssuerAndWallet('did:ethr:0x1234', WalletType.ANY);`

Please see supported wallet types in `src/types/wallet-types.ts`.

### ✅ Fixed Issues

- [List any fixed issues here like: Fixes #XXXX]

### 🚨 Test instructions

`npm test`

### ⚠️ Don't forget to add a [semver](https://semver.org/) label!

- `patch`: Bug Fix?
- `minor`: New Feature?
- `major`: Breaking Change?
- `skip-release`: It's unnecessary to publish this change.

### CODE REVIEW

File: src/modules/users/index.ts [hunk 1]
```diff
@@ -34,24 +62,16 @@ export class UsersModule extends BaseModule {
       email: string | null;
       oauth_provider: string | null;
       phone_number: string | null;
-    }>(`${this.sdk.apiBaseUrl}/v1/admin/auth/user/get`, this.sdk.secretApiKey, { issuer });
+      wallets: MagicWallet[] | null;
+    }>(`${this.sdk.apiBaseUrl}/v1/admin/auth/user/get?wallet_type=${walletType}`, this.sdk.secretApiKey, { issuer });
```
Comment (dgerrellsMagic) [inline src/modules/users/index.ts:66]: What is the default wallet type? I am thinking of if this breaks existing integrations which assume ethereum.

File: test/spec/modules/users/getMetadataByPublicAddress.spec.ts [hunk 2]
```diff
@@ -11,3 +12,15 @@ test('#01: Successfully GETs to metadata endpoint via public address', async ()
   const getMetadataArguments = getMetadataStub.mock.calls[0];
   expect(getMetadataArguments).toEqual(['did:ethr:0x1234']);
 });
+
+test('#02: Successfully GETs to metadata endpoint via public address and wallet type', async () => {
```
Comment (dgerrellsMagic) [inline test/spec/modules/users/getMetadataByPublicAddress.spec.ts:16]: While the pattern here of `#2` for test cases exists today it is a pretty bad standard. This is fine for now but test names should stand alone without the numbering. 


Could we add negative tests? What happens if I pass in a nonexistent chain? or just something totally bad.

File: src/modules/users/index.ts (hunk 1 above)
Comment (justinnout) [reply src/modules/users/index.ts:66]: Default type is WalletType.None which will default to the existing behavior of `GET /v1/admin/auth/user/get`.

File: test/spec/modules/users/getMetadataByPublicAddress.spec.ts (hunk 2 above)
Comment (justinnout) [reply test/spec/modules/users/getMetadataByPublicAddress.spec.ts:16]: Got it, I can remove the numbers from all the tests.

Each of the new functions require an argument of type WalletType so clients can only pass in chains we support. I have tests for SOLANA, and ANY but I can add one for NONE. There are negative tests for bad / unexpected inputs in DID parsing logic which we are reusing.

**Review by dgerrellsMagic — APPROVED** [review]

**Review by dgerrellsMagic — APPROVED** [review]

**Pull request merged**